server/drivers/hd44780-gpio.c

```diff
@@ -159,12 +175,12 @@ hd_init_gpio(Driver *drvthis)
 
 	ugpio_set_value(pins->rs, 0);
 
-	send_nibble(p, (FUNCSET | IF_8BIT) >> 4);
+	send_nibble(p, (FUNCSET | IF_8BIT) >> 4, 0);
 	p->hd44780_functions->uPause(p, 4100);
-	send_nibble(p, (FUNCSET | IF_8BIT) >> 4);
+	send_nibble(p, (FUNCSET | IF_8BIT) >> 4, 0);
 	p->hd44780_functions->uPause(p, 100);
-	send_nibble(p, (FUNCSET | IF_8BIT) >> 4);
-	send_nibble(p, (FUNCSET | IF_4BIT) >> 4);
+	send_nibble(p, (FUNCSET | IF_8BIT) >> 4, 0);
+	send_nibble(p, (FUNCSET | IF_4BIT) >> 4, 0);
 
```

Comment:
I think you are missing initialization of the second display here?
(I don't know anything about the protocol or the initialization procedure - just applying common sense, so maybe I'm wrong.)